freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/downscale/handler/CollectAdditionalHostnamesHandler.java

```diff
@@ -0,0 +1,67 @@
+package com.sequenceiq.freeipa.flow.freeipa.downscale.handler;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.IpaServer;
+import com.sequenceiq.freeipa.flow.freeipa.downscale.event.DownscaleFailureEvent;
+import com.sequenceiq.freeipa.flow.freeipa.downscale.event.collecthostnames.CollectAdditionalHostnamesRequest;
+import com.sequenceiq.freeipa.flow.freeipa.downscale.event.collecthostnames.CollectAdditionalHostnamesResponse;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class CollectAdditionalHostnamesHandler implements EventHandler<CollectAdditionalHostnamesRequest> {
```

Comment:
`EventHandler` is deprecated, could you switch to `ExceptionCatcherEventHandler`?